static/js/services/lineage-model-generator.js

```diff
@@ -29,16 +30,47 @@ angular.module('inboxServices').factory('LineageModelGenerator',
         });
     };
 
+    var hydrate = function(lineage) {
```

Comment:
the hydrate functions are multiplying like rabbits!! 🐰 🐰 🐰 
Filed https://github.com/medic/medic-webapp/issues/3826